docs/single_member_restoration.md

```diff
@@ -0,0 +1,16 @@
+## Restoration of a single member in multi-node etcd deployed by etcd-druid.
+
+**Note**:
+- For a cluster with n members, we are proposing the solution to only single member restoration within a etcd cluster not the quorum loss scenario (when majority of members within a cluster fail).
+- In this proposal we are not targetting the recovery of single member which got separated from cluster due to [network partition](https://etcd.io/docs/v3.3/op-guide/failures/#network-partition).
+
+## Motivation
+If a single etcd member within a multi-node etcd cluster goes down due to DB corruption/PVC corruption/Invalid data-dir then it needs to be brought back. Unlike in the single-node case, a minority member of a multi-node cluster can't be restored from the snapshots present in storage container as you can't restore from the old snapshots as it contains the metadata information of cluster which leads to **memberID mismatch** that prevents the new member from coming up as new member is getting it's metadata information from db which got restore from old snapshots.
+
+## Solution
+- Corresponding backup-restore sidecar if detects that its corresponding etcd is down due to [data-dir corruption](https://github.com/gardener/etcd-backup-restore/blob/7d27a47f5793b0949492d225ada5fd8344b6b6a2/pkg/initializer/validator/datavalidator.go#L177) or [Invalid data-dir](https://github.com/gardener/etcd-backup-restore/blob/7d27a47f5793b0949492d225ada5fd8344b6b6a2/pkg/initializer/validator/datavalidator.go#L204)
+- Then, if it's a cluster with cluster size `>1` then backup-restore will first remove the etcd failed member using the [MemberRemove API](https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/clientv3/cluster.go#L45-L46) call and clean the data-dir of failed etcd member.
+- It won't affect the etcd cluster as quorum is still maintained.
```

Comment:
So lets take the following case:
There are 3 etcd members - `etcd-1`, `etcd-2` and `etcd-3`.
There was data corruption in `etcd-2` - back-restore triggers the process of cleaning the data dir, removing the member and promoting the member. While this is being done, there is another member for which data corruption happens, back-up restore for that pod also tries the same thing. It removes the member but when it attempts to add the member as a learner it fails as you can have atmost one learner at a time. How do you propose to handle this situation? Also will Abhishek's proposal of quorum loss kick in here which yet another actor (etcd-druid) handles?

Reply:
> add the member as a learner it fails as you can have atmost one learner at a time.

no, this will fail as there is no quorum present not because of `one learner at a time`.
And for quorum loss:
yes, I'm assuming Abhishek's proposal of quorum loss should take care of this as I mentioned in doc as well that .... `this proposal doesn’t take care of quorum loss`.